replace_attr_test.go

```diff
@@ -0,0 +1,52 @@
+package slogx_test
+
+import (
+	"log/slog"
+	"testing"
+	"time"
+
+	"github.com/powerman/check"
+
+	"github.com/powerman/slogx"
+)
+
+func TestReplaceAttr(tt *testing.T) {
+	t := check.T(tt)
+
+	var (
+		id     = "ID"
+		userID = "UserID"
+
+		modifyAttrValue = func(_ []string, a slog.Attr) slog.Attr {
+			if a.Key == id {
+				a.Value = slog.StringValue("REDACTED")
+			}
+			return a
+		}
+		modifyAttrKey = func(_ []string, a slog.Attr) slog.Attr {
+			if a.Key == id {
+				a.Key = userID
+			}
+			return a
+		}
+		returnZeroAttr = func(_ []string, a slog.Attr) slog.Attr {
+			if a.Key == slog.TimeKey {
+				return slog.Attr{}
+			}
+			return a
+		}
+		ignoreAfterZeroAttr = func(_ []string, a slog.Attr) slog.Attr {
+			if a.Key == slog.TimeKey {
+				a.Value = slog.AnyValue(time.Now().Round(time.Hour))
+			}
+			return a
+		}
+	)
+
+	t.Panic(func() { slogx.ChainReplaceAttr() })
+
+	fn := slogx.ChainReplaceAttr(modifyAttrValue, modifyAttrKey, returnZeroAttr, ignoreAfterZeroAttr)
+	t.DeepEqual(fn([]string{}, slog.Attr{Key: id, Value: slog.IntValue(325)}), slog.Attr{Key: userID, Value: slog.StringValue("REDACTED")})
+	t.DeepEqual(fn([]string{"g"}, slog.Attr{Key: id, Value: slog.IntValue(325)}), slog.Attr{Key: userID, Value: slog.StringValue("REDACTED")})
```

Comment:
I added the call with not empty group. But I am not sure you mean exactly such test. Group gets lost. @powerman

Reply:
Groups does not lost, they just didn't affect chained function's output. Test should check is chained functions will get same groups in their argument.